### PR TITLE
Made OCM optional and temporarily disabled

### PR DESCRIPTION
**Pull request by mateusz-garlacz**

### CODE REVIEW

File: cs3api4lab/api/share_api_facade.py [hunk 1]
```diff
@@ -39,7 +39,10 @@ def __init__(self, log):
     def create(self, endpoint, file_path, opaque_id, idp, role=Role.EDITOR, grantee_type=Grantee.USER, reshare=True):
         """Creates a share or creates an OCM share if the user is not found in local domain"""
         if self._is_ocm_user(opaque_id, idp):
-            return self.ocm_share_api.create(opaque_id, idp, idp, endpoint, file_path, grantee_type, role, reshare)
+            if self.config['enable_ocm']:
+                return self.ocm_share_api.create(opaque_id, idp, idp, endpoint, file_path, grantee_type, role, reshare)
+            else:
```
Comment (diocas) [inline cs3api4lab/api/share_api_facade.py:44]: We should fail as this request should not reach us. `501`? 
We should do this for all other cases where it applies such an error.

**Review by diocas — CHANGES_REQUESTED** [review]
Please add errors instead of just silently failing. The frontend needs to be aware this is optional and handle attempts from users of sharing with OCM.
Please also rebase.

**Comment by dagl** [comment]
I think we should extend this functionality project-wise, for example in cs3_user_api:find_user_by_query we also try to use ocm

**Comment by diocas** [comment]
You also try OCM when searching for a users, but you only if it's enabled. I was expecting this PR to fix all those cases (the web part, maybe can come in a 2nd PR).

**Comment by mateusz-garlacz** [comment]
> You can also try OCM when searching for a users, but you only if it's enabled. I was expecting this PR to fix all those cases (the web part, maybe can come in a 2nd PR).

I updated the PR. What do you mean by web part ?

File: cs3api4lab/api/cs3_user_api.py [hunk 2]
```diff
@@ -50,12 +50,13 @@ def find_users_by_query(self, query):
         ocm_response = self.invite_api.FindAcceptedUsers(ia.FindAcceptedUsersRequest(filter=query),
```
Comment (diocas) [inline cs3api4lab/api/cs3_user_api.py:50]: Shouldn't this be inside the if as well?

**Review by diocas — COMMENTED** [review]
Just one last comment and we can merge.

What I meant about the web is that it should not receive the errors you just added. Those are for anyone trying to circumvent the API. The web should prevent any attempt to use OCM if it's not available. I don't know if that's already the case, but I would expect it to be (for example, if searching for a user the OCM options do not appear, it's expectable that they are not used when creating a share so all is good). So this was just a remark :)

**Comment by diocas** [comment]
@mateusz-garlacz when you removed the ocm handlers you forgot to remove the import and LGTM is now complaining. (and thanks for removing the handlers, now we really have a unified api!)

**Comment by mateusz-garlacz** [comment]
Ah sorry, my bad. Now I made lgtm happy

**Pull request merged**